pkg/services/sqlstore/sqlstore.go

```diff
@@ -185,7 +186,11 @@ func (ss *SqlStore) buildConnectionString() (string, error) {
 
 		cnnstr += ss.buildExtraConnectionString('&')
 	case migrator.POSTGRES:
-		host, port := util.SplitHostPortDefault(ss.dbCfg.Host, "127.0.0.1", "5432")
+		host, port, err := util.SplitHostPortDefault(ss.dbCfg.Host, "127.0.0.1", "5432")
+		if err != nil {
+			return "", xerrors.Errorf("Invalid host specifier '%s': %w", ss.dbCfg.Host, err)
```

Comment:
We have some utils for working with xerrors that I would suggest to use. See https://github.com/grafana/grafana/blob/master/pkg/util/errutil/errors.go#L23